asmcomp/schedgen.ml

```diff
@@ -44,18 +44,21 @@ let dummy_node =
    - code_checkbounds contains the latest checkbound node not matched
      by a subsequent load or store. *)
 
-let code_results = (Hashtbl.create 31 : (location, code_dag_node) Hashtbl.t)
-let code_uses = (Hashtbl.create 31 : (location, code_dag_node) Hashtbl.t)
-let code_stores = ref ([] : code_dag_node list)
-let code_loads = ref ([] : code_dag_node list)
-let code_checkbounds = ref ([] : code_dag_node list)
-
-let clear_code_dag () =
-  Hashtbl.clear code_results;
-  Hashtbl.clear code_uses;
-  code_stores := [];
-  code_loads := [];
-  code_checkbounds := []
+type code_dag =
+  { results : (location, code_dag_node) Hashtbl.t
+  ; uses : (location, code_dag_node) Hashtbl.t
+  ; mutable stores : code_dag_node list
+  ; mutable loads : code_dag_node list
+  ; mutable checkbounds : code_dag_node list
+  }
```

Comment:
For the record, I think that this is one of the worse formatting styles for records (types and expressions), and it is also not the one currently used in the file. I would support silently breaking consistency by moving to my own favorite style, but I feel slightly grumpy about this one. (But it's not very important and I think you should leave it as-is if you really care about this.)

The desired property for my own formatting choices is that we can add or reorder lines, and the resulting diff only highlights the relevant lines and not the others. Your proposed style does not have this property for the first item, which is different from others (this is arguably better than the style in this file, which breaks the property for both the first and the last line; the last line is more important). It is also non-standard in the OCaml codebase so more likely to raise eyebrows.

Reply:
fixed. I'm used to the leading ';' style and keep forgetting to not use it in the compiler.